code/controllers/ContentController.php

```diff
@@ -87,42 +88,59 @@ public function Page($link) {
 
 	public function init() {
 		parent::init();
-		
+
+		$recordExists = ($this->dataRecord && $this->dataRecord->exists());
+
 		// If we've accessed the homepage as /home/, then we should redirect to /.
-		if($this->dataRecord && $this->dataRecord instanceof SiteTree
-			 	&& RootURLController::should_be_on_root($this->dataRecord) && (!isset($this->urlParams['Action']) || !$this->urlParams['Action'] ) 
-				&& !$_POST && !$_FILES && !$this->redirectedTo() ) {
+		if(
+			$recordExists
+			&& RootURLController::should_be_on_root($this->dataRecord)
+			&& (!isset($this->urlParams['Action']) || !$this->urlParams['Action'])
+			&& !$_POST
+			&& !$_FILES
```

Comment:
Could we just check `$this->request->postVars()`? IIRC `$_FILES` is merged into `$_POST` for this. Appreciate you’re just tidying, not refactoring though ;)